src/components/ColorPicker.tsx

```diff
@@ -111,18 +178,10 @@ const ColorSwatch: FC<{
 const ColorPicker: FC<{ fontSize: number; style?: React.CSSProperties }> = ({ fontSize, style }) => {
   const colors = useSelector(themeColors)
   const ref = useRef<HTMLDivElement>(null)
-  const cursorStyle = useSelector(
-    state =>
-      state.showColorPicker && state.cursor
-        ? {
-            // merge =style (which contains color) and =styleAnnotation (which contains backgroundColor)
-            // the style attribute name is not relevant here
-            ...(getStyle(state, head(state.cursor)) || ({} as React.CSSProperties)),
-            ...getStyle(state, head(state.cursor), { attributeName: '=styleAnnotation' }),
-          }
-        : undefined,
-    shallowEqual,
-  )
+  const cursorStyle = {
```

Comment:
This reference is not stable, causing the props to ColorSwatch to change on every render, and preventing ColorSwatch from being memoized.

I recommend defining `backgroundColor` and `color` as separate variables and then using `useMemo` to construct a stable ref.